mainapp/admin.py

```diff
@@ -49,47 +48,49 @@ class VolunteerAdmin(admin.ModelAdmin):
     readonly_fields = ('joined',)
     list_display = ('name', 'phone', 'organisation', 'joined')
     list_filter = ('district', 'joined',)
+
     def download_csv(self, request, queryset):
         options = vol_categories
         mapper = {}
         for i in vol_categories:
             mapper[i[0]] = i[1] 
-        f = open('vol.csv', 'w')
-        writer = csv.writer(f)
-        l = []
-        for i in (Volunteer._meta.get_fields()):
-            l.append(i.name)
-        writer.writerow(l)
-        data = Volunteer.objects.all().values_list()
-        for s in data:
-            s = list(s)
-            s[6] = mapper[s[6]]
-            writer.writerow(s)
-        f.close()
-        f = open('vol.csv', 'r')
-        response = HttpResponse(f, content_type='text/csv')
-        response['Content-Disposition'] = 'attachment; filename=Volunteers.csv'
+        with open('vol.csv', 'w') as file_:
+            writer = csv.writer(file_)
+            l = []
```

Comment:
Could have used a list comprehension here and better not to use single lettered variable name

Reply:
``volunteer_list = [vol.name for vol in Volunteer._meta.get_fields()]``
``writer.writerow(volunteer_list)``

The similar usages repeats in below code. They are better than loops.